framework/assets/yii.validation.js

```diff
@@ -436,5 +436,32 @@ yii.validation = (function ($) {
         }
     }
 
+    function validateMimeType(mimeTypes, fileType)
+    {
+        for (var i=0, len = mimeTypes.length; i<len; i++) {
+            if (mimeTypes[i] === fileType) {
+                return true;
+            }
+
+            if (mimeTypes[i].indexOf('*') !== -1 && validateMimeTypeMask(mimeTypes[i], fileType)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    function validateMimeTypeMask(mimeTypeMask, fileType)
+    {
+        var escapedString = escapeRegExp(mimeTypeMask);
+        var regexp = new RegExp('^' + escapedString.replace('\\*', '.*') +  '$');
+
+        return regexp.test(fileType);
+    }
+
+    function escapeRegExp(str) {
+        return (str+'').replace(/[.?*+^$[\]\\(){}|-]/g, "\\$&");
```

Comment:
@SilverFire didn't get that, Client side doesn't receive regex from PHP side, it's generating it's own using mimeType mask